default-placeholders/index.md

```diff
@@ -0,0 +1,269 @@
+A default constructed placeholder accessor is analogous to a null pointer.
+A placeholder that's bound to a buffer
+but hasn't been registered with a command group
+is more like a fancy pointer:
+the user doesn't own the data
+until the accessor is registered and used in a kernel,
+where is becomes more similar to a regular pointer.
+
+Having this type separation between full accessors and placeholders
+might be useful from a type safety perspective,
+but we believe it makes development more difficult.
+For example, [another one of our proposals](https://github.com/codeplaysoftware/standards-proposals/pull/100/files)
+introduces alias templates for different access targets
+and revises rules on how read-only accessors are handled,
+all in the name of reducing accessor verbosity.
+The placeholder template parameter makes that much more difficult,
+meaning we either need to introduce another parameter to the alias template,
+making it a lot less useful,
+or simply ignore the reduction in verbosity for placeholder accessors.
+
+## Changes
+
+### Deprecate `access::placeholder`
+
+Mark the `access::placeholder` enum class as deprecated,
+but keep it for backwards compatibility
+until it's eventually removed from a subsequent standard.
+
+```cpp
+namespace access {
+...
+
+enum class placeholder // deprecated
+{...};
+} // namespace access
+
+template <typename dataT,
+          int dimensions,
+          access::mode accessMode,
+          access::target accessTarget = access::target::global_buffer,
+          access::placeholder isPlaceholder = access::placeholder::false_t>
+class accessor;
+```
+
+### All accessors with a global or constant target can be placeholders
+
+SYCL 1.2.1 allows `access::placeholder::true_t`
+only when the access target is `global_buffer` or `constant_buffer`.
+We propose that the same placeholder semantics
+still apply to only these two targets,
+just that the semantics and API is available
+regardless of the `isPlaceholder` template parameter.
+
+### Accept all accessors in `handler::require`
+
+At the moment the member function `handler::require` only accepts
+placeholder accessors.
+This should stay the same,
+but since the enum class is deprecated,
+the function needs to be extended:
+
+```cpp
+class handler {
+ public:
+  ...
+
+  // Adds isPlaceholder to existing SYCL 1.2.1 function
+  template <typename dataT,
+            int dimensions,
+            access::mode accessMode,
+            access::target accessTarget,
+            access::placeholder isPlaceholder>
+  void require(accessor<dataT,
+                        dimensions,
+                        accessMode,
+                        accessTarget,
+                        isPlaceholder>
+                acc);
+};
+```
+
+### Deprecate `is_placeholder`
+
+The function `accessor::is_placeholder` doesn't make sense anymore,
+we propose deprecating it.
+
+### New constructors
+
+We propose adding new constructors to the accessors class
```

Comment:
We may also need to make a tweak to the `buffer::get_access`  variant which takes no parameters, as that's currently defined to always return an `accessor` of `access::target::host_buffer`. If we want to be able to retrieve a placeholder `accessor` using this member function as well, we will need to disambiguate it. One option could be to require that the `access::target` be specified, another could be to have a different member function for this. Alternatively, we could just rely on the `accessor` constructor for this, as `get_access`` is simply syntactic sugar over the constructor.

Reply:
I'm not sure `get_access` should return a placeholder, it's probably better to always return a full accessor. In terms of verbosity, https://github.com/codeplaysoftware/standards-proposals/pull/100 would probably be sufficient for resolving that.

Reply:
I think this is a good approach, this means that `get_access` is the full contract of associating the `accessor` with a `buffer` and `handler`, and if you want to create an `accessor` from a `buffer` without associating a `handler` with it you just use the constructor.